drivers/gpio/gpio_mchp_xec.c

```diff
@@ -170,26 +250,81 @@ static int gpio_xec_write(struct device *dev,
 
 }
 
-static int gpio_xec_read(struct device *dev,
-			 int access_op, u32_t pin, u32_t *value)
+static int gpio_xec_port_set_masked_raw(struct device *dev, u32_t mask,
+					u32_t value)
+{
+	const struct gpio_xec_config *config = dev->config->config_info;
+
+	/* GPIO output registers are used for writing */
+	__IO u32_t *gpio_base = GPIO_OUT_BASE(config);
+
+	*gpio_base = (*gpio_base & ~mask) | (mask & value);
+
+	return 0;
+}
+
+static int gpio_xec_port_set_bits_raw(struct device *dev, u32_t mask)
+{
+	const struct gpio_xec_config *config = dev->config->config_info;
+
+	/* GPIO output registers are used for writing */
+	__IO u32_t *gpio_base = GPIO_OUT_BASE(config);
+
+	*gpio_base |= mask;
+
+	return 0;
+}
+
+static int gpio_xec_port_clear_bits_raw(struct device *dev, u32_t mask)
+{
+	const struct gpio_xec_config *config = dev->config->config_info;
+
+	/* GPIO output registers are used for writing */
+	__IO u32_t *gpio_base = GPIO_OUT_BASE(config);
+
+	*gpio_base &= ~mask;
+
+	return 0;
+}
+
+static int gpio_xec_port_toggle_bits(struct device *dev, u32_t mask)
+{
+	const struct gpio_xec_config *config = dev->config->config_info;
+
+	/* GPIO output registers are used for writing */
+	__IO u32_t *gpio_base = GPIO_OUT_BASE(config);
+
+	*gpio_base ^= mask;
+
+	return 0;
+}
+
+static int gpio_xec_port_get_raw(struct device *dev, u32_t *value)
 {
 	const struct gpio_xec_config *config = dev->config->config_info;
-	u32_t port_n = config->port_num;
 
 	/* GPIO input registers are used for reading */
-	__IO u32_t *gpio_base = (__IO u32_t *)(GPIO_PARIN_BASE + (port_n << 2));
+	__IO u32_t *gpio_base = GPIO_IN_BASE(config);
 
 	*value = *gpio_base;
+
+	return 0;
+}
+
+static int gpio_xec_read(struct device *dev,
+			 int access_op, u32_t pin, u32_t *value)
+{
+	int ret;
+
+	ret = gpio_xec_port_get_raw(dev, value);
+
 	if (access_op == GPIO_ACCESS_BY_PIN) {
 		*value = (*value & BIT(pin)) >> pin;
-	} else { /* GPIO_ACCESS_BY_PORT not supported */
-		return -EINVAL;
 	}
 
-	return 0;
+	return ret;
```

Comment:
The `gpio_xec_port_get_raw` function always returns 0, still it looks like we are changing the legacy behavior. Let's leave it as `return 0;`.